plugins/formal-verification/formver.plugin/src/org/jetbrains/kotlin/formver/reporting/VerifierErrorInterpreter.kt

```diff
@@ -99,4 +106,40 @@ class VerifierErrorInterpreter {
             is SourceRole.ReturnsEffect.Null -> if (negated) "null" else "non-null"
             else -> TODO("Unreachable")
         }
+
+    private fun SourceRole.ReturnsEffect.describe(): String = when (this) {
+        is SourceRole.ReturnsEffect.Bool -> "a $bool value is returned"
+        is SourceRole.ReturnsEffect.Null -> "a ${if (negated) "non-null" else "null"} value is returned"
+        SourceRole.ReturnsEffect.Wildcard -> "the function returns"
+    }
```

Comment:
Can this be unified with the `ReturnsEffect` logic?

Reply:
You mean moving the inside `ReturnsEffect` definition?

Reply:
Using the same code to get messages for `ReturnsEffect` in both cases, if it's possible, so you don't have to process them in two places.  It's not blocking, though.

Reply:
Ah yeah! Then I have an idea.